.github/workflows/cd.yml

```diff
@@ -1,10 +1,18 @@
 name: cd
 
-on: [push, pull_request, workflow_dispatch]
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+    workflow_dispatch:
```

Comment:
I think this is not valid YAML. If we can't put anything here, we can remove `workflow_dispatch`.

Reply:
According to the reference, it seems to be valid: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#using-activity-types-and-filters-with-multiple-events

I believe workflow_dispatch falls under the category of an event without configuration

Reply:
Ok. Let's see if this works.